### DISCUSSION TITLE
Confused about syncing the changed props with the context

### DISCUSSION THREAD

**Discussion by nightire** [statelyai/xstate#3478] [Q&A] [answered]
Let's say if `context.foo` get an initial value from the component's `props.foo`, what's the strategy to update the `context.foo` if `props.foo` has changed?

I'm afraid this "update context with changed props" idea is an anti-pattern or, worse, stupid idea. However, I could not figure out a way to see this through.

Take a regular `<Input />` component as an example:

```jsx
const Input = ({ disabled = false }: InputProps) => {
  const [state] = useMachine(inputMachine, { context: { disabled } });
  return  <input disabled={state.matches('disabled')} />;
};
```

It's natural to hold the `disabled` in a machine's context, but it doesn't reflect the changes of the `props.disabled`. It's unclear to me to get the correct model.

I set up a code example to elaborate on my confusion: https://codesandbox.io/s/nice-nova-tpi6o8

**Comment by Andarist** [comment, marked as answer]
You can do this:
```js
useEffect(() => {
  send({ type: 'DISABLED_CHANGED', disabled })
}, [disabled])
```

**Reply by nightire** [reply]
Oh, I tried to solve this inside the machine, never thought about using `useEffect`. I'll try that, thank you!

**Reply by nightire** [reply]
I think I made it work as I expected, in case anyone cares, I updated my demo here: https://codesandbox.io/s/nice-nova-tpi6o8
